### DISCUSSION TITLE
cell_annotation

### DISCUSSION THREAD

**Discussion by Chhiring-Lama** [welch-lab/MultiVelo#14] [Q&A]
Hi, 
I am using multivelo for my snATAC and snRNA data. I started by following the demo you had on [10x data](https://github.com/welch-lab/MultiVelo/blob/main/multivelo_demo/MultiVelo_Demo.ipynb). There at a point we have to supply cell_annotaion.tsv. Is it the metadata of the seurat object we fed to produce the loom file from velocyto?

**Comment by danielee0707** [comment]
I'm not exactly sure what you mean. The provided cell_annotation.tsv is for the mouse brain demo dataset only. If you're using your own dataset, you can annotate the clusters directly based on Scanpy or Seurat marker genes. If you have a list of barcodes and corresponding cell types, you can indeed make one cell_annotation.tsv using the provided one as a template.
I don't think Velocyto needs a metadata file when generating loom files.

**Comment by AnaAzambuja** [comment]
Hi... another question question for the cell_annotation. 
I am running with my dataset and I generated a cell_annotation.tsv file using yours as template.
When I run it, i getting this:

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[36], line 1
----> 1 adata_rna = adata_rna[cell_annot.index,:]
      2 adata_rna.obs['celltype'] = cell_annot['celltype']

File ~/miniconda3/envs/py38Ana/lib/python3.8/site-packages/anndata/_core/anndata.py:1113, in AnnData.__getitem__(self, index)
   1111 def __getitem__(self, index: Index) -> "AnnData":
   1112     """Returns a sliced view of the object."""
-> 1113     oidx, vidx = self._normalize_indices(index)
   1114     return AnnData(self, oidx=oidx, vidx=vidx, asview=True)

File ~/miniconda3/envs/py38Ana/lib/python3.8/site-packages/anndata/_core/anndata.py:1094, in AnnData._normalize_indices(self, index)
   1093 def _normalize_indices(self, index: Optional[Index]) -> Tuple[slice, slice]:
-> 1094     return _normalize_indices(index, self.obs_names, self.var_names)

File ~/miniconda3/envs/py38Ana/lib/python3.8/site-packages/anndata/_core/index.py:35, in _normalize_indices(index, names0, names1)
     33         index = index[0].values, index[1]
     34 ax0, ax1 = unpack_index(index)
---> 35 ax0 = _normalize_index(ax0, names0)
     36 ax1 = _normalize_index(ax1, names1)
     37 return ax0, ax1

File ~/miniconda3/envs/py38Ana/lib/python3.8/site-packages/anndata/_core/index.py:101, in _normalize_index(indexer, index)
     99         if np.any(positions < 0):
    100             not_found = indexer[positions < 0]
--> 101             raise KeyError(
    102                 f"Values {list(not_found)}, from {list(indexer)}, "
    103                 "are not valid obs/ var names or indices."
    104             )
    105         return positions  # np.ndarray[int]
    106 else:

KeyError: "Values ['AAACCGAAGAGGGACT-1', 'AAACGCGCAGCATGAG-1', 'AACTAGCTCTCACTCA-1', 'AACTTAGTCGCACACA-1', 'AAGAACAGTGATGGCT-1', 'AAGGATCCATCTTGAG-1'....TTTGTTGGTGCATTTC-1'], are not valid obs/ var names or indices."


Any idea?
